tests/utils/test_filters.py

```diff
@@ -29,19 +31,101 @@ def data():
 def test_spatial_uniform_filter_3d(data):
     """Test spatial_uniform_filter_3d with SpatialDimension."""
 
-    res = spatial_uniform_filter_3d(data, SpatialDimension(3, 3, 3))
-    assert torch.sum(res) == torch.sum(data)
+    result = spatial_uniform_filter_3d(data, SpatialDimension(3, 3, 3))
+    assert data.shape == result.shape
+    assert torch.isclose(torch.sum(result), torch.sum(data))
 
 
 def test_spatial_uniform_filter_3d_tuple(data):
     """Test spatial_uniform_filter_3d with tuple."""
 
-    res = spatial_uniform_filter_3d(data, (3, 3, 3))
-    assert torch.sum(res) == torch.sum(data)
+    result = spatial_uniform_filter_3d(data, (3, 3, 3))
+    assert data.shape == result.shape
+    assert torch.isclose(torch.sum(result), torch.sum(data))
 
 
 def test_spatial_uniform_filter_wrong_width(data):
     """Test spatial_uniform_filter_3d with wrong width."""
 
     with pytest.raises(ValueError, match='Invalid filter width'):
         spatial_uniform_filter_3d(data, (3, 3))
+
+
+def test_gaussian_filter_int_axis(data):
+    result = gaussian_filter(data, 0.5, -1)
+    assert result.nonzero() == torch.tensor(
+        [[0, 0, 2, 2, 0], [0, 0, 2, 2, 1], [0, 0, 2, 2, 2], [0, 0, 2, 2, 3], [0, 0, 2, 2, 4]]
+    )
+    assert result.shape == data.shape
+    torch.testing.assert_close(result.sum(), torch.tensor(1.0), atol=1e-3, rtol=0)
+
+
+def test_gaussian_filter_two_axis(data):
+    result = gaussian_filter(data, 0.5, (-1, 3))
+    assert result.shape == data.shape
+    torch.testing.assert_close(result.sum(), torch.tensor(1.0), atol=1e-3, rtol=0)
+
+
+def test_gaussian_filter_two_sigmas(data):
+    result = gaussian_filter(data, (0.2, 0.5), (-1, 3))
+    assert result.shape == data.shape
+    torch.testing.assert_close(result.sum(), torch.tensor(1.0), atol=1e-3, rtol=0)
+
+
+def test_gaussian_filter_noaxis(data):
```

Comment:
is it worth adding a test to ensure that the results along the different axis match? e.g. result[0,0,:,2,2] == result[0,0,2,:,2] == ... ?